File: Week_02/G20190343020253/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        times = dict()
        for i in range(0, len(s)):
            s_alpha = s[i]
            if s_alpha in times:
                times[s_alpha] += 1
            else:
                times[s_alpha] = 1
        for i in range(0, len(t)):
            t_alpha = t[i]
            if t_alpha in times:
                times[t_alpha] -= 1
                if times[t_alpha] == 0:
                    times.pop(t_alpha)
            else:
                times[t_alpha] = 1
        if times:
            return False
        else:
            return True

File: Week_02/G20190343020023/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    var S, T [26]int32
    for _, v := range s {
        S[v - 'a']++
    }
    for _, v := range t {
        T[v - 'a']--
    }
    
    for i := 0; i < len(S); i++ {
        if S[i] + T[i] != 0 {
            return false
        }
    }
    return true
}